messenger.rst

```diff
@@ -78,78 +62,74 @@ message handler. It's a class with an ``__invoke`` method::
-
-.. configuration-block::
+.. code-block:: terminal
 
-    .. code-block:: yaml
+    $ php bin/console debug:messenger
 
-        # config/services.yaml
-        services:
-            App\MessageHandler\SmsNotificationHandler:
-                tags: [messenger.message_handler]
+Dispatching the Message
+-----------------------
 
-    .. code-block:: xml
+You're ready! To dispatch the message (and call the handler), inject the
+``message_bus`` service (via the ``MessageBusInterface``), like in a controller::
 
-        <!-- config/services.xml -->
-        <?xml version="1.0" encoding="UTF-8" ?>
-        <container xmlns="http://symfony.com/schema/dic/services"
-            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-            xsi:schemaLocation="http://symfony.com/schema/dic/services
-                https://symfony.com/schema/dic/services/services-1.0.xsd">
+    // src/Controller/DefaultController.php
+    namespace App\Controller;
 
-            <services>
-                <service id="App\MessageHandler\SmsNotificationHandler">
-                   <tag name="messenger.message_handler"/>
-                </service>
-            </services>
-        </container>
+    use App\Message\SmsNotification;
+    use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+    use Symfony\Component\Messenger\MessageBusInterface;
 
-    .. code-block:: php
+    class DefaultController extends AbstractController
+    {
+        public function index(MessageBusInterface $bus)
+        {
+            // will cause the SmsNotificationHandler to be called
+            $bus->dispatch(new SmsNotification('Look! I created a message!'));
 
-        // config/services.php
-        use App\MessageHandler\SmsNotificationHandler;
+            // or use the shortcut
+            $this->dispatchMessage(new SmsNotification('Look! I created a message!'));
 
-        $container->register(SmsNotificationHandler::class)
-            ->addTag('messenger.message_handler');
+            // ...
+        }
+    }
 
-.. note::
+Transports: Async/Queued Messages
+---------------------------------
 
-    If the message cannot be guessed from the handler's type-hint, use the
-    ``handles`` attribute on the tag.
+By default, messages are handled as soon as they are dispatched. If you want
+to handle a message asynchronously, you can configure a transport. A transport
+is capable of sending messages (e.g. to a queueing system) and then receiving
+them via a worker. Messenger supports :ref:`multiple transports <messenger-transports-config>`.
 
-Transports
-----------
+.. note::
 
-By default, messages are processed as soon as they are dispatched. If you prefer
-to process messages asynchronously, you must configure a transport. These
-transports communicate with your application via queuing systems or third parties.
+    If you need more message brokers, check out `Enqueue's transport`_,
```

Comment:
I think linking to the adapter is best - that's precisely what the user needs to look at next. Then it would be the job of the adapter's docs to direct the user through the rest of the process, including other links :)